src/zenml/cli/hub.py

```diff
@@ -0,0 +1,1203 @@
+        version: Version of the plugin. If not provided, the latest version
+            will be used.
+    """
+    with event_handler(
+        event=AnalyticsEvent.ZENML_HUB_PLUGIN_LOGS,
+        metadata={
+            "plugin_name": plugin_name,
+            "plugin_version": version,
+        },
+    ) as analytics_handler:
+        client = HubClient()
+        analytics_handler.metadata["hub_url"] = client.url
+        plugin_name, author = _parse_plugin_name(plugin_name)
+        analytics_handler.metadata["plugin_name"] = plugin_name
+        analytics_handler.metadata["plugin_author"] = author
+        display_name = _plugin_display_name(plugin_name, version, author)
+
+        # Get the plugin from the hub
+        plugin = client.get_plugin(
+            plugin_name=plugin_name, plugin_version=version, author=author
+        )
+        if not plugin:
+            error(f"Could not find plugin '{display_name}' on the hub.")
+        analytics_handler.metadata["plugin_version"] = plugin.version
+
+        if plugin.status == PluginStatus.PENDING:
+            error(
+                f"Plugin '{display_name}' is still being built. Please try "
+                "again later."
+            )
+
+        if not plugin.build_logs:
+            logger.info(
+                f"Plugin '{display_name}' finished building, but no logs "
+                "were found."
+            )
+            return
+
+        for line in plugin.build_logs.splitlines():
+            if line.startswith("DEBUG"):
+                pass
+            if line.startswith("INFO"):
+                logger.info(line)
+            elif line.startswith("WARNING"):
+                logger.warning(line)
+            else:
+                logger.error(line)
+
+
+# GENERAL HELPER FUNCTIONS
+
+
+def _is_plugin_installed(plugin: HubPluginResponseModel) -> bool:
+    """Helper function to check if a plugin is installed.
+
+    Args:
+        plugin: The plugin.
+
+    Returns:
+        Whether the plugin is installed.
+    """
+    module_name = _get_plugin_module(plugin)
+    try:
+        spec = find_spec(module_name)
+        return spec is not None
+    except ModuleNotFoundError:
+        return False
+
+
+def _get_plugin_module(plugin: HubPluginResponseModel) -> str:
+    """Helper function to get the module name of a plugin.
+
+    Args:
+        plugin: The plugin.
+
+    Returns:
+        The module name of the plugin.
+    """
+    module_name = "zenml.hub"
+    username = _get_plugin_author_username(plugin)
+    if username:
+        module_name += f".{username}"
+    module_name += f".{plugin.name}"
+    return module_name
+
+
+def _get_plugin_author_username(plugin: HubPluginResponseModel) -> str:
+    """Helper function to get the username of the author of a plugin.
+
+    Args:
+        plugin: The plugin.
+
+    Returns:
+        The username of the plugin author or "" if no username was found.
+    """
+    if plugin.user:
```

Comment:
That's what we thought initially, but actually, it can never be `None` now since ZenML official plugins also have an admin user account associated with it. This still needs to be fixed both here and in the hub, thanks for pointing out 👍

Reply:
The model now no longer has `plugin.user` on it, but instead a non-optional `plugin.author` which is the username of the user that created the plugin.